test/dashboardHandler.test.js

```diff
@@ -0,0 +1,369 @@
+'use strict';
+
+var AWS = require('aws-sdk-mock');
+var awsSdk = require('aws-sdk');
+var chai = require("chai");
+var sinonChai = require("sinon-chai");
+chai.use(sinonChai);
+var expect = chai.expect;
+var LambdaTester = require('lambda-tester');
+var index = require('../src/dashboardHandler');
+var sinon = require('sinon');
+
+describe("generateDashboardTrend", () => {
+  let listMetricsStub;
+  let putDashboardSpy;
+  var sandbox = sinon.createSandbox();
+
+  function generateMetrics(n) {
+    return [...Array(n).keys()].map(idx => {
+      return [{
+        "Namespace": "Operations",
+        "Dimensions": [
+          {
+            "Name": "service",
+            "Value": `service-${idx}`
+          }
+        ],
+        "MetricName": "MTTR"
+      },
+      {
+        "Namespace": "Operations",
+        "Dimensions": [
+          {
+            "Name": "service",
+            "Value": `service-${idx}`
+          }
+        ],
+        "MetricName": "MTBF"
+      },
+      {
+        "Namespace": "Operations",
+        "Dimensions": [
+          {
+            "Name": "service",
+            "Value": `service-${idx}`
+          }
+        ],
+        "MetricName": "MTTF"
+      }
+      ]
+    });
+  }
+  let scenarios = [
+    {
+      description: "sample dashboard - single service",
+      dashboard: "./sample-data/dashboard-sample.json",
+      uniqueServices: 1,
+      metrics: {
+        "Metrics": [
+          {
+            "Namespace": "Operations",
+            "Dimensions": [
+              {
+                "Name": "service",
+                "Value": "flaky-service"
+              }
+            ],
+            "MetricName": "MTTR"
+          },
+          {
+            "Namespace": "Operations",
+            "Dimensions": [
+              {
+                "Name": "service",
+                "Value": "flaky-service"
+              }
+            ],
+            "MetricName": "MTTF"
+          },
+          {
+            "Namespace": "Operations",
+            "Dimensions": [
+              {
+                "Name": "service",
+                "Value": "flaky-service"
+              }
+            ],
+            "MetricName": "MTBF"
+          }
+        ]
+      },
+      event: {
+        "account": "123456789012",
+        "region": "ap-southeast-2",
+        "detail": {},
+        "detail-type": "Scheduled Event",
+        "source": "aws.events",
+        "time": "2019-03-01T01:23:45Z",
+        "id": "cdc73f9d-aea9-11e3-9d5a-835b769c0d9c",
+        "resources": [
+          "arn:aws:events:ap-southeast-2:123456789012:rule/my-schedule"
+        ]
+      }
+    }, {
+      description: "sample dashboard - multiple service",
+      dashboard: "./sample-data/dashboard-multiple-sample.json",
+      uniqueServices: 2,
+      metrics: {
+        "Metrics": [
```

Comment:
Is there value in refactoring this to use `[].concat.apply([], generateMetrics(2))`?

Reply:
Done